java/memory/src/main/java/io/netty/buffer/ArrowBuf.java

```diff
@@ -558,134 +529,138 @@ private void ensureWritable(final int length) {
+   * @param writerIndex index to start write data
    * @param value value to set
    */
-  public void writeByte(byte value) {
-    ensureWritable(1);
+  public void writeByte(int writerIndex, byte value) {
+    ensureWritable(writerIndex,1);
     PlatformDependent.putByte(addr(writerIndex), value);
-    ++writerIndex;
   }
 
   /**
    * Set the lower order byte for the provided value at
    * the writerIndex.
+   * @param writerIndex index to start write data
    * @param value value to be set
    */
-  public void writeByte(int value) {
-    ensureWritable(1);
+  public void writeByte(int writerIndex, int value) {
+    ensureWritable(writerIndex,1);
     PlatformDependent.putByte(addr(writerIndex), (byte)value);
-    ++writerIndex;
   }
 
   /**
    * Write the bytes from given byte array into this
    * ArrowBuf starting at writerIndex.
+   * @param writerIndex index to start write data
    * @param src src byte array
    */
-  public void writeBytes(byte[] src) {
+  public void writeBytes(int writerIndex, byte[] src) {
     Preconditions.checkArgument(src != null, "expecting valid src array");
-    writeBytes(src, 0, src.length);
+    writeBytes(writerIndex, src, 0, src.length);
   }
 
   /**
    * Write the bytes from given byte array starting at srcIndex
    * into this ArrowBuf starting at writerIndex.
+   * @param writerIndex index to start write data
    * @param src src byte array
    * @param srcIndex index in the byte array where the copy will being from
    * @param length length of data to copy
    */
-  public void writeBytes(byte[] src, int srcIndex, int length) {
-    ensureWritable(length);
+  public void writeBytes(int writerIndex, byte[] src, int srcIndex, int length) {
+    ensureWritable(writerIndex, length);
     setBytes(writerIndex, src, srcIndex, length);
-    writerIndex += length;
   }
 
   /**
    * Set the provided int value as short at the writerIndex.
+   * @param writerIndex index to start write data
    * @param value value to set
    */
-  public void writeShort(int value) {
-    ensureWritable(SHORT_SIZE);
+  public void writeShort(int writerIndex, int value) {
+    ensureWritable(writerIndex, SHORT_SIZE);
     PlatformDependent.putShort(addr(writerIndex), (short) value);
-    writerIndex += SHORT_SIZE;
   }
 
   /**
    * Set the provided int value at the writerIndex.
+   * @param writerIndex index to start write data
    * @param value value to set
    */
-  public void writeInt(int value) {
-    ensureWritable(INT_SIZE);
+  public void writeInt(int writerIndex, int value) {
+    ensureWritable(writerIndex, INT_SIZE);
     PlatformDependent.putInt(addr(writerIndex), value);
-    writerIndex += INT_SIZE;
   }
 
   /**
    * Set the provided long value at the writerIndex.
+   * @param writerIndex index to start write data
    * @param value value to set
    */
-  public void writeLong(long value) {
-    ensureWritable(LONG_SIZE);
+  public void writeLong(int writerIndex, long value) {
```

Comment:
I think we should simply remove these methods. Once you have index they are effectively the same as set

Reply:
Hmm, I did not noticed this, fixed now, thanks!